src/Entity/ItemId.php

```diff
@@ -23,8 +24,9 @@ class ItemId extends EntityId implements Int32EntityId {
 	 * @throws InvalidArgumentException
 	 */
 	public function __construct( $idSerialization ) {
-		$this->assertValidIdFormat( $idSerialization );
-		$this->serialization = strtoupper( $idSerialization );
+		parent::__construct( self::upcaseLocalId( $idSerialization ) );
```

Comment:
Why is upcaseLocalId called here, when both upcaseLocalId and the constructor call are in the parent class? The way this code is organized is a bit confusing and can mislead people into using this wrong. Basically: Not all entity IDs must be upper case. This is a decision the specific subclass (ItemId and PropertyId) must make. Which raises the question why the helper method is in the base class. Answer: Code sharing.

In the past we tried hard to get rid of code sharing in the Entity and EntityId base classes.

I suggest to rearrange the code in a way that you can reuse the split method, do a trivial `strtoupper( $array[2] )` here in the subclass and the pass the array to the base class.